arrow-optics/src/test/kotlin/arrow/optics/StackOverflowTest.kt

```diff
@@ -0,0 +1,98 @@
+package arrow.optics
+
+import arrow.core.Either
+import arrow.data.ListKW
+import arrow.data.ListKWKind
+import arrow.data.k
+import arrow.syntax.either.left
+import arrow.syntax.either.right
+import arrow.test.UnitSpec
+import org.assertj.core.api.Assertions.assertThat
```

Comment:
Can we use kotlin-test assertions such as `shouldBe` to not add this extra dependency?, Kotlin test is already in the classpath

Reply:
Done